tests/helpers.go

```diff
@@ -0,0 +1,59 @@
+package tests
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+
+	. "github.com/onsi/ginkgo"
+)
+
+func ChangeDirectory(dir string) {
+	err := os.Chdir(dir)
+	if err != nil {
+		Fail("Failed to change directory: " + err.Error())
+	}
+}
+
+func CreateTemplate() string {
+	input := []byte("test-gin-templates")
+
+	reader, writer, err := os.Pipe()
+	if err != nil {
+		Fail("Failed to connected pair of files: " + err.Error())
+	}
+
+	_, err = writer.Write(input)
+	if err != nil {
+		Fail("Failed to write file: " + err.Error())
+	}
+	writer.Close()
+
+	shCmd := exec.Command("cookiecutter", "../..")
```

Comment:
This should actually use the command `nimble-gin create` as would any user do. It's critical to use the built-in command.

Reply:
Yes it should but `nimble-gin create` doesn't allow us to test on current working branch but only default git branch because `create` command executes `cookiecutter https://github.com/nimblehq/gin-templates` which use source code from default git branch.

I cannot think of the best way to do this. Do you have any idea?

Reply:
The branch name should be a parameter in the command and default to the master branch if it's omitted: 

![image](https://user-images.githubusercontent.com/696529/108466356-13660700-72b6-11eb-84b5-836120255cad.png)

cookie-cutter supports specifying a branch (see [docs](https://cookiecutter.readthedocs.io/en/1.7.2/usage.html#works-directly-with-git-and-hg-mercurial-repos-too)):

![image](https://user-images.githubusercontent.com/696529/108466397-24167d00-72b6-11eb-8ce2-ccb531a2d4e8.png)

Reply:
Fixed in d5b18bec99962908241262f937ed088039a1e860 c5e2d4ddce83a3b60fc5edad31db906366c77eb9

Reply:
To test if the application template has new tests.

- Checkout to `test-template-unit-test` branch.
- Run `BRANCH=develop make test`. This should fail because template test cannot find `test-specifying-branch.yml`
- Run `BRANCH=test-template-unit-test make test`. This should pass.